sdk/typescript/src/cryptography/publickey.ts

```diff
@@ -1,136 +1,82 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
-
 import BN from 'bn.js';
-import { Buffer } from 'buffer';
-import { sha3_256 } from 'js-sha3';
 
 /**
- * Value to be converted into public key
+ * Value to be converted into public key.
  */
-export type PublicKeyInitData =
-  | number
-  | string
-  | Buffer
-  | Uint8Array
-  | Array<number>
-  | PublicKeyData;
+ export type PublicKeyInitData =
+ | number
+ | string
+ | Buffer
+ | Uint8Array
+ | Array<number>
+ | PublicKeyData;
 
 /**
- * JSON object representation of PublicKey class
- */
+* JSON object representation of PublicKey class.
+*/
 export type PublicKeyData = {
-  /** @internal */
-  _bn: BN;
+ /** @internal */
+ _bn: BN;
 };
 
-export const PUBLIC_KEY_SIZE = 32;
-export const TYPE_BYTE = 0x00;
-
+/**
+ * A keypair used for signing transactions.
+ */
 export type SignatureScheme = 'ED25519' | 'Secp256k1';
-
-const SIGNATURE_SCHEME_TO_FLAG = {
+ 
+export const SIGNATURE_SCHEME_TO_FLAG = {
   ED25519: 0x00,
   Secp256k1: 0x01,
 };
 
-function isPublicKeyData(value: PublicKeyInitData): value is PublicKeyData {
+export function checkPublicKeyData(value: PublicKeyInitData): value is PublicKeyData {
```

Comment:
this is renamed due to name collision with index.guard.ts for another isPublicKeyData